### PR TITLE
[Breaking] Unify CoinType values 

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes https://github.com/trustwallet/wallet-core/issues/1927

Breaking changes:

1. `TWCoinType` matches `id` in `registry.json`

### CoinType.swift
- `bandChain` -> `band`
- `bitcoinCash` -> `bitcoincash`
- `bitcoinGold` -> `bitcoingold`
- `digiByte` -> `digibyte`
- `dogecoin` -> `doge`
- `smartChain` -> `smartchain`
- `ecochain` -> `heco`
- `avalancheCChain` -> `avalanchec`
- `ethereumClassic` -> `classic`
- `goChain` -> `gochain`
- `ioTeX` -> `iotex`
- `poanetwork` -> `poa`
- `thunderToken` -> `thundertoken`
- `tomoChain` -> `tomochain`
- `veChain` -> `vechain`
- `cryptoOrg` -> `cryptoorg`

### wallet.core.jni.CoinType

- `ETHEREUMCLASSIC` -> `CLASSIC`
- `ECOCHAIN` -> `HECO`
- `AVALANCHECCHAIN` -> `AVALANCHEC`
- `POANETWORK` -> `POA`
- `DOGECOIN` -> `DOGE`
- `BANDCHAIN` -> `BAND`

2. Generated lowercase Swift enum values

- `Blockchain.swift`
- `PublicKeyType.swift`
- `EthereumChainID.swift`
- `HRP.swift`

3. `Blockchain` value is lowercase in  `registry.json`

4. Other changes:
- newcoin script + template updated
- xDai -> Gnosis Chain

### CODE REVIEW

File: docs/registry.md [hunk 1]
```diff
@@ -48,7 +47,7 @@ This list is generated from [./registry.json](../registry.json)
 | 500     | Theta            | THETA  | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/theta/info/logo.png" width="32" />        | <https://www.thetatoken.org>  |
 | 501     | Solana           | SOL    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/solana/info/logo.png" width="32" />       | <https://solana.com>          |
 | 508     | Elrond           | eGLD   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/elrond/info/logo.png" width="32" />       | <https://elrond.com/>         |
-| 714     | Binance          | BNB    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/binance/info/logo.png" width="32" />      | <https://binance.org>         |
+| 714     | BNB              | BNB    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/binance/info/logo.png" width="32" />      | <https://binance.org>         |
```
Comment (catenocrypt) [inline docs/registry.md:50]: Binance would be more appropriate here, not?  Display name is BNB.

File: include/TrustWalletCore/TWCoinType.h [hunk 2]
```diff
@@ -29,15 +29,15 @@ enum TWCoinType {
     TWCoinTypeBitcoinCash = 145,
     TWCoinTypeBitcoinGold = 156,
     TWCoinTypeCallisto = 820,
-    TWCoinTypeCardano = 1815, // Note: Cardano Shelley testnet uses purpose 1852 (not 44) 1852/1815
+    TWCoinTypeCardano = 1815,
```
Comment (catenocrypt) [inline include/TrustWalletCore/TWCoinType.h:32]:
```suggestion
    TWCoinTypeCardano = 1815, // Note: Cardano Shelley testnet uses purpose 1852 (not 44) 1852/1815
```

File: samples/osx/cocoapods/WalletCoreExample.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (catenocrypt) [inline samples/osx/cocoapods/WalletCoreExample.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: Is this file really needed in repo?

File: docs/registry.md [hunk 4]
```diff
@@ -18,14 +18,13 @@ This list is generated from [./registry.json](../registry.json)
...
 | 145     | Bitcoin Cash     | BCH    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/bitcoincash/info/logo.png" width="32" />  | <https://bitcoincash.org>     |
 | 148     | Stellar          | XLM    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/stellar/info/logo.png" width="32" />      | <https://stellar.org>         |
 | 156     | Bitcoin Gold     | BTG    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/bitcoingold/info/logo.png" width="32" />  | <https://bitcoingold.org>     |
 | 165     | Nano             | XNO    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/nano/info/logo.png" width="32" />         | <https://nano.org>            |
 | 175     | Ravencoin        | RVN    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ravencoin/info/logo.png" width="32" />    | <https://ravencoin.org>       |
-| 178     | POA Network      | POA    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/poa/info/logo.png" width="32" />          | <https://poa.network>         |
```
Comment (catenocrypt) [inline docs/registry.md:28]: POA is removed from registry.md, is this right?

**Review by catenocrypt — APPROVED** [review]

**Comment by catenocrypt** [comment]
DisplayName for Binance Chain is `BNB`, wouldn't Binance be better?

File: docs/registry.md (hunk 1 above)
Comment (hewigovens) [reply docs/registry.md:50]: BNB was the branded name, `registry.md.erb` not updated

File: include/TrustWalletCore/TWCoinType.h (hunk 2 above)
Comment (hewigovens) [reply include/TrustWalletCore/TWCoinType.h:32]: This comment looks confusing to me, we won't support its testnet anyway; if you really need it, maybe put it somewhere else?

File: docs/registry.md (hunk 4 above)
Comment (hewigovens) [reply docs/registry.md:28]: Yeah, because https://www.poa.network will be merged into xDai

Comment (catenocrypt) [reply docs/registry.md:28]: OK, I got confused as POA is not being removed from 'registry.json', and I did not notice that it is being 'dperecated' there. +1

**Comment by hewigovens** [comment]
@vikmeup take a look if you have any chance, what else can be improved
